controllers/helpers.go

```diff
@@ -211,7 +177,11 @@ func getEtcdMachineAddress(machine *clusterv1.Machine) string {
 }
 
 func getMemberClientURL(address string) string {
-	return fmt.Sprintf("https://%s:2379", address)
+	return fmt.Sprintf("%s%s:%s", httpsPrefix, address, etcdClientURLPort)
+}
+
+func getEtcdMachineAddressFromClientURL(clientURL string) string {
+	return strings.TrimSuffix(strings.TrimPrefix(clientURL, httpsPrefix), fmt.Sprintf(":%s", etcdClientURLPort))
```

Comment:
nit: parsing the url and getting the `.Host` would probably be a bit cleaner and maybe more performant